corehq/apps/reports/tasks.py

```diff
@@ -298,16 +299,29 @@ def build_form_multimedia_zip(domain, xmlns, startdate, enddate, app_id,
     num_forms = len(forms_info)
     DownloadBase.set_progress(build_form_multimedia_zip, 0, num_forms)
 
+    case_id_to_name = _get_case_names(
+        domain,
+        set.union(*map(lambda form_info: form_info['case_ids'], forms_info)),
+    )
+
     use_transfer = settings.SHARED_DRIVE_CONF.transfer_enabled
     if use_transfer:
         fpath = _get_download_file_path(xmlns, startdate, enddate, export_id, app_id, num_forms)
     else:
         _, fpath = tempfile.mkstemp()
 
-    _write_attachments_to_file(fpath, use_transfer, num_forms, forms_info)
+    _write_attachments_to_file(fpath, use_transfer, num_forms, forms_info, case_id_to_name)
     _expose_download(fpath, use_transfer, zip_name, download_id, num_forms)
 
 
+def _get_case_names(domain, case_ids):
+    case_id_to_name = {c: c for c in case_ids}
+    for case in CaseAccessors(domain).iter_cases(case_ids):
```

Comment:
actually will be one query per export. which makes me wonder if it might be worth keeping it as `iter_` cause we may end up loading a lot of cases into memory if we use ES. unless we implement an ES scroll